code/game/machinery/boss.dm

```diff
@@ -0,0 +1,130 @@
+//B.O.S.S.: Bluespace Object Support System
+
+//contains:
+//the main BOSS sender and variants
+//the BOSS receiver
+
+//TODO
+//clean up lists when receivers are destroyed?
+//make them require power
+//make them drop held items when destroyed
+
+//BOSS sender//
+/obj/machinery/boss_sender
+	name = "\improper B.O.S.S. sender"
+	desc = "The Bluespace Object Support System, or 'B.O.S.S.' is used to teleport items to far away places to aid others."
+	icon_state = "autolathe"
+
+	density = TRUE
+	anchored = TRUE
+
+	obj_integrity = 250
+	max_integrity = 250
+
+	var/checkTech = FALSE
+
+	var/multiple_receivers = TRUE
+
+	var/obj/machinery/boss_receiver/linked
+
+/obj/machinery/boss_sender/sci
+	desc = "The Bluespace Object Support System, or 'B.O.S.S.' is used to teleport equipment made in R&D to lavaland, to aid the miners. Interact with an open hand to send or remove items."
+	checkTech = TRUE
+	multiple_receivers = FALSE
+
+/obj/machinery/boss_sender/sci/Initialize()
+	..()
+	for(var/obj/machinery/boss_receiver/B in GLOB.boss_receivers)
+		if(B.is_mining_receiver)
+			linked = B
+			return
+
+/obj/machinery/boss_sender/attack_hand(mob/user)
+	..()
+	if(powered())
+		switch(alert(user,"Select an option.","[src.name] panel","Send all items","Manage inventory"))

```

Comment:
Alert/input on machinery as their UI is really bad. Do a browser datum like remie said.